src/main/java/org/broadinstitute/hellbender/tools/exome/BayesianHetPulldownCalculator.java

```diff
@@ -0,0 +1,671 @@
+
+        double errorLogLikelihood = 0.0;
+        double refAltLogLikelihood;
+        ArrayList<Double> alphaList = new ArrayList<>();
+        ArrayList<Double> betaList = new ArrayList<>();
+
+        for (Nucleotide base : baseQualities.keySet()) {
+            if (base == alleleRef) {
+                for (BaseQuality currentBaseQuality : baseQualities.get(base)) {
+                    alphaList.add(currentBaseQuality.getReadErrorProbability() / 3 +
+                        currentBaseQuality.getMappingErrorProbability() / 4);
+                    betaList.add(1 - 4 * currentBaseQuality.getReadErrorProbability() / 3 -
+                        currentBaseQuality.getMappingErrorProbability() / 4);
+                }
+            }
+            else if (base == alleleAlt) {
+                for (BaseQuality currentBaseQuality : baseQualities.get(base)) {
+                    alphaList.add(1 - currentBaseQuality.getReadErrorProbability() -
+                        3 * currentBaseQuality.getMappingErrorProbability() / 4);
+                    betaList.add(-1 + 4 * currentBaseQuality.getReadErrorProbability() / 3 +
+                        currentBaseQuality.getMappingErrorProbability());
+                }
+            }
+            else {
+                for (BaseQuality currentBaseQuality : baseQualities.get(base)) {
+                    errorLogLikelihood += getSafeguardedLogProbability(currentBaseQuality.getReadErrorProbability() / 3 +
+                        currentBaseQuality.getMappingErrorProbability() / 4);
+                }
+            }
+        }
+
+        switch (hetPriorType) {
+
+            case BALANCED:
+                /* set the allele fraction to 1/2 */
+                refAltLogLikelihood = getRefAltHetLogLikelihoodFixedAlleleFraction(0.5, alphaList, betaList);
+                break;
+
+            case HETEROGENEOUS:
+                /* marginalize allele fraction by integrating {@link getRefAltHetLogLikelihoodFixedAlleleFraction} over
+                   the precomputed allele fraction prior */
+                ArrayList<Double> refAltLogLikelihoodList = new ArrayList<>(gaussIntegrationAbscissas.size());
+                refAltLogLikelihoodList.addAll(
+                        gaussIntegrationAbscissas.stream()
+                                .mapToDouble(Double::doubleValue)
+                                .map(f -> getRefAltHetLogLikelihoodFixedAlleleFraction(f, alphaList, betaList))
+                                .boxed().collect(Collectors.toList()));
+                ArrayList<Double> logLikelihoodIntegrandWithPriorAndWeights = new ArrayList<>(gaussIntegrationAbscissas.size());
+                logLikelihoodIntegrandWithPriorAndWeights.addAll(
+                        IntStream.range(0, gaussIntegrationAbscissas.size())
+                                .mapToDouble(i -> refAltLogLikelihoodList.get(i) + gaussIntegrationLogWeights.get(i) +
```

Comment:
Actually, `.mapToDouble(Double::doubleValue).map(X).boxed()`  can be replaced with `.map(X)`. I don't remember what I had in mind when I wrote that in the first place.